fuse_optimizers/src/variable_stamp_index.cpp

```diff
@@ -120,6 +134,12 @@ void VariableStampIndex::applyRemovedVariables(const fuse_core::Transaction& tra
 {
   for (const auto& variable_uuid : transaction.removedVariables())
   {
+    // remove from unique stamps
+    auto stamp = stamped_index_[variable_uuid];
+    if(unique_stamps_.find(stamp) != unique_stamps_.end()){
+      unique_stamps_.erase(stamp);
+    }
```

Comment:
This comment is important:
Multiple variables may be associated with the same timestamp. The motion model in the fuse_models package uses position, orientation, linear velocity, angular velocity, and linear acceleration variables all at the exact same timestamp. Just because you remove the linear acceleration variable doesn't mean that the other variables are also removed. It feels like you really need to keep track of the individual variables associated with each timestamp, and only remove a timestamp once all variables have been removed.
Some sort of structure like:
```
using StampToVariablesMap = std::map<ros::Time, std::unordered_set<fuse_core::UUID>>;
StampToVariablesMap unique_stamps_;
```